src/persist-client/src/internal/machine.rs

```diff
@@ -1355,6 +1317,59 @@ pub mod datadriven {
         Ok(format!("parts={} len={}\n", batch.parts.len(), batch.len))
     }
 
+    pub async fn write_rollup(
```

Comment:
might not be super obvious from the diff: I've split out `write-rollup` (writing a rollup blob into Blob) and `add-rollup` (linking said rollup into Consensus). This matches how we handle batches in datadriven tests, and allows us to consolidate on the `write_rollup_for_state` method as the only way to write a batch.